pkg/manager/manager.go

```diff
@@ -176,15 +176,10 @@ func (m *Manager) doDaemonFailover(d *daemon.Daemon) {
 	}
 
 	// Failover nydusd still depends on the old supervisor
-
-	if err := m.StartDaemon(d); err != nil {
-		log.L.Errorf("fail to start daemon %s when recovering", d.ID())
-		return
-	}
+	m.StartDaemonUntilSubscribed(d)
 
 	if err := d.WaitUntilState(types.DaemonStateInit); err != nil {
-		log.L.WithError(err).Errorf("daemon didn't reach state %s,", types.DaemonStateInit)
-		return
+		log.L.WithError(err).Errorf("daemon didn't reach state %s", types.DaemonStateInit)
```

Comment:
If the daemon fails to reach `Init` state, we can't invoke `takeOver` API, it conflicts the nydusd internal state machine